zipline/pipeline/factors/technical.py

```diff
@@ -144,6 +148,205 @@ def compute(self, today, assets, out, close, volume):
         out[:] = nanmean(close * volume, axis=0)
 
 
+class CorrelationFactor(SingleInputMixin, CustomFactor):
+    """
+    Correlation of the given asset with all other assets for the given input.
+
+    **Default Inputs:** None
+
+    **Default Window Length:** None
+
+    Parameters
+    ----------
+    inputs : length-1 list/tuple of BoundColumn
+        The expression over which to compute the correlations.
+    window_length : int > 1
+        Length of the look back window over which to compute correlations.
+    asset : zipline.assets.Asset
+        The asset to correlate with all other assets.
+
+    Example
+    -------
+    Let the following be example close prices for three different stocks::
+
+                      AAPL   MSFT     FB
+        2017-03-13       5      6      1
+        2017-03-14       5      4      2
+        2017-03-15       4      9      3
+        2017-03-16       6      7      4
+        2017-03-17       3      4      5
+        2017-03-20       2      6      6
+        2017-03-21       2      5      7
+        2017-03-22       2      8      8
+
+    Suppose we are interested in AAPL's rolling close price correlation with
+    each stock from 2017-03-17 to 2017-03-22, using a 5-day look back window
+    (that is, we calculate each correlation coefficient over 5 days of data).
+    We can achieve this by doing::
+
+        correlation_factor = CorrelationFactor(
+            inputs=[USEquityPricing.close],
+            window_length=5,
+            asset=Equity(24),
+        )
+
+    The result of computing ``correlation_factor`` from 2017-03-17 to
+    2017-03-22 gives::
+
+                      AAPL   MSFT     FB
+        2017-03-17       1    .21   -.42
+        2017-03-20       1    .15   -.70
+        2017-03-21       1    .51   -.76
+        2017-03-22       1    .18   -.82
+
+    Note that the column for AAPL is all 1's, as the correlation of any data
+    with itself is always 1. To understand how each of the other values were
+    calculated, take for example the .21 in MSFT's column. This is the Pearson
+    correlation coefficient between AAPL's five close prices looking back from
+    2017-03-17 (5, 5, 4, 6, 3) and MSFT's five close prices (6, 4, 9, 7, 4).
+
+    See Also
+    --------
+    :class:`zipline.pipeline.factors.technical.SingleRegressionFactor`
+    """
+    params = ['asset']
+
+    def compute(self, today, assets, out, data, asset):
+        asset_col = where(assets == asset.sid)[0][0]
+        # Use the transpose here because we want the correlations between
```

Comment:
I think you can just pass `rowvar=0`:

```
rowvar : int, optional
    If `rowvar` is non-zero (default), then each row represents a
    variable, with observations in the columns. Otherwise, the relationship
    is transposed: each column represents a variable, while the rows
    contain observations.
```